assets/src/data/model/entity-factory/base-entity.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * External imports
+ */
+import { isArray, upperFirst, camelCase } from 'lodash';
+
+/**
+ * Internal imports
+ */
+import { assertValidSchema } from './assertions';
+import {
+	createGetter,
+	createEntityGettersAndSetters,
+	createPersistingGettersAndSetters,
+	setSaveState,
+} from './create';
+import {
+	SAVE_STATE,
+	PRIVATE_PROPERTIES,
+} from './constants';
+
+/**
+ * BaseEntity is the basic class for all entities.  createEntityFactory returns
+ * an instance of this and all the getters/setters for fields etc are
+ * dynamically created via the constructor.
+ */
+class BaseEntity {
```

Comment:
This is basically the common parent for all our entity objects and handles dynamically setting up getters/setters from the incoming schema from the model.